provisioner/template.go

```diff
@@ -733,3 +734,49 @@ func sumQuantities(quantities ...string) (string, error) {
 func awsValidID(id string) string {
 	return strings.Replace(id, ":", "__", -1)
 }
+
+// divideInstanceResourcesInRatio takes in an instance type and a ratio and returns the result of applying
+// the ratio on the instance CPU and Memory sizes.
+// The ratio must be a positive number and less than 1.
+// The resourceType must be either 'cpu' or 'memory'.
+// The result is rounded down to the nearest integer, to avoid overcommitting resources.
+// The units for CPU and memory are millicores and mebibytes (MiB) respectively.
+func divideInstanceResourcesInRatio(adapter *awsAdapter, instanceType string, resourceType string, ratio float32) (dividedResource int, err error) {
```

Comment:
I'll check. Thanks!

Reply:
Please also consider adding a set of functions:
1. a pair of `instanceCpu` and `instanceMemory` receiving single instance type argument and returns cpu/memory as a k8s quantity
2. `scaleQuantity(value, factor)` returns value*factor as a k8s quantity.

This way we get a set of composable functions that would allow e.g. original scenario via:
```
scaleQuantity (instanceCpu "m4.xlarge") 0.9
```
or fixed-size offset using existing `sumQuantities`:
```
sumQuantities (instanceMemory "m4.xlarge") "-200Mi"
```

Reply:
I've added the functions in the latest commit, haven't replaced the AWS API calls with the in-memory data yet.

Reply:
Also, I think as a template function, `scaleQuantity` should return the `Quantity` as `String` no? So that it can be used directly in the template. I'm basing this on the fact that `sumQuantities` does the same.

Reply:
I think all of them should work with (receive as args and return) string representation of quantity. This would allow passing string const string arguments e.g. `scaleQuantity "100Mi" 0.7`

The cost of parsing it internally should be negligible.

Reply:
I agree. I'll update them to string representations. Then we should be done. Will then add unit tests and that should be it.

Reply:
Update in the latest commit.